### PR TITLE
Support gitlab tools

### PR DESCRIPTION
**Pull request by jeparlefrancais**
Hello👋 

I wanted to give out this implementation I did to be able to download tooling I maintain on Gitlab, like [darklua](https://gitlab.com/seaofvoices/darklua).

### CODE REVIEW

**Comment by jeparlefrancais** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/main.rs [hunk 1]
```diff
@@ -139,30 +156,54 @@ fn actual_main() -> io::Result<()> {
...
     }
 
     Ok(())
 }
+
+fn prompt_auth_token(token: Option<String>, provider: &str, help: &str) -> io::Result<String> {
+    match token {
+        Some(token) => Ok(token),
+        None => {
+            println!("GitHub auth saved successfully.");
```
Comment (ZoteTheMighty) [inline src/main.rs]: This wording needs to use the `provider` arg, right?

**Review by ZoteTheMighty — APPROVED** [review]

File: src/main.rs (hunk 1 above)
Comment (jeparlefrancais) [reply src/main.rs]: Correct ✔️

**Review by ZoteTheMighty — APPROVED** [review]
Looks good!

**Review by oltrep — APPROVED** [review]

**Pull request merged**